src/mono/mono/mini/aot-compiler.c

```diff
@@ -14355,6 +14382,50 @@ add_interp_in_wrapper_for_sig (MonoAotCompile *acfg, MonoMethodSignature *sig)
 	add_method (acfg, wrapper);
 }
 
+static gboolean
+mono_aot_add_direct_pinvoke (MonoAotCompile *acfg, char *dpi)
+{
+	char *direct_pinvoke = g_strdup (dpi);
+	if (!direct_pinvoke)
+		return TRUE;
+
+	if (direct_pinvoke[0] == '#')
+		return TRUE;
+
+	g_strstrip (direct_pinvoke);
+	if (!direct_pinvoke)
+		return TRUE;
+
+	char **direct_pinvoke_split = g_strsplit (direct_pinvoke, "!", -1);
+	char *library_name = g_strdup (direct_pinvoke_split[0]);
+	if (!direct_pinvoke_split[1]) {
+		g_hash_table_insert (acfg->direct_pinvokes, library_name, NULL);
+		g_strfreev (direct_pinvoke_split);
+		return TRUE;
+	}
+	if (direct_pinvoke_split[2]) {
+		aot_printerrf (acfg, "The provided 'direct_pinvoke' AOT option '%s' shouldn't contain multiple '!'\n", dpi);
+		g_strfreev (direct_pinvoke_split);
+		return FALSE;
+	}
+
+	char *entrypoint_name = g_strdup (direct_pinvoke_split[1]);
+	if (g_hash_table_contains (acfg->direct_pinvokes, library_name)) {
```

Comment:
Combine the two calls into:

GHashTable *val;
if (g_hash_table_lookup_extended (acfg->direct_pinvokes, library_name, NULL, &val))
...

Reply:
Thanks!